src/Core/Scheduler/Providers/WMS_Provider.js

```diff
@@ -57,6 +67,16 @@ WMS_Provider.prototype.preprocessDataLayer = function preprocessDataLayer(layer)
     layer.style = layer.style || '';
     layer.transparent = layer.transparent || false;
 
+    // add tileMatrixSet destination to tile
```

Comment:
I still think that this doesn't belong to WMS_Provider.

Let me know if I understand correctly: the info you're adding here are helping to convert extent from the geometry to extent(s) in the layer's crs?

In which case maybe it would make sense to move this to `TileMesh.getCoordsForLayer`... WDYT?